jb/20210103/split_of_child_share_fixed.py

```diff
@@ -0,0 +1,362 @@
+
+  # Percent winners
+  target_persons['better_off'] = (target_persons.new_spm_resources > target_persons.spm_resources)
+  total_better_off = (target_persons.better_off * target_persons.person_weight).sum()
+  percent_better_off = total_better_off / pop
+
+  return pd.Series([poverty_rate, gini, poverty_gap, percent_better_off, adult_ubi, child_ubi])
+
+def ubi2(funding_billions=0, child_percent_ubi=0): 
+  funding = funding_billions * 1e9
+  child_percent_ubi /=100
+  adult_ubi = (funding / (adult_pop + (child_pop * child_percent_ubi)))
+  child_ubi = adult_ubi * child_percent_ubi
+
+  tax_rate = funding / total_taxable_income
+
+  spmu['new_tax'] = tax_rate * spmu.tax_inc
+  spmu['spm_ubi'] = (spmu.child * child_ubi) + (spmu.adult * adult_ubi)
+
+  spmu['new_spm_resources'] = spmu.spm_resources + spmu.spm_ubi - spmu.new_tax
+  spmu['new_spm_resources_pp'] = spmu.new_spm_resources / spmu.spm_numper
+
+  # Calculate poverty gap
+  poverty_gap = pov_gap(spmu, 'new_spm_resources', 'spm_povthreshold', 'spm_weight')
+
+  # Merge person and spmu dataframes
+  spmu_sub = spmu[['spm_id', 'new_spm_resources', 'new_spm_resources_pp']]
+  target_persons = pd.merge(spmu_sub,person,on=['spm_id'])
+
+  target_persons['poor'] = target_persons.new_spm_resources < target_persons.spm_povthreshold
+  total_poor = (target_persons.poor * target_persons.person_weight).sum()
+  poverty_rate = (total_poor / pop * 100)
+
+  # Calculate Gini
+  gini = mdf.gini(target_persons, 'new_spm_resources_pp', w='person_weight')
+
+  # Percent winners
+  target_persons['better_off'] = (target_persons.new_spm_resources > target_persons.spm_resources)
+  total_better_off = (target_persons.better_off * target_persons.person_weight).sum()
+  percent_better_off = total_better_off / pop
+
+  return pd.Series([poverty_rate, gini, poverty_gap, percent_better_off, adult_ubi, child_ubi])
+
+# create a dataframe with all possible combinations of funding levels and
+summary = mdf.cartesian_product({'funding_billions': np.arange(0,3_001,50),
+                                 'child_percent_funding': np.arange(0, 101, 1)})
+
+def ubi_row(row):  
+    return ubi(row.funding_billions, row.child_percent_funding)
+summary[['poverty_rate', 'gini', 'poverty_gap', 'percent_better_off', 'adult_ubi', 'child_ubi']] = summary.apply(ubi_row, axis=1)
+
+summary['monthly_child_ubi'] =summary['child_ubi'].apply(lambda x: int(round(x/12,0)))
+summary['monthly_adult_ubi'] =summary['adult_ubi'].apply(lambda x: int(round(x/12,0)))
+summary.to_csv("children_share_funding_summary.csv.gz",compression='gzip')
+
+# drop rows where funding level is 0
+optimal_poverty = summary.sort_values('poverty_gap').drop_duplicates('funding_billions', keep='first')
+optimal_poverty = optimal_poverty.drop(
+    optimal_poverty[optimal_poverty.funding_billions==0].index
+    ) 
+
+optimal_inequality = summary.sort_values('gini').drop_duplicates('funding_billions', keep='first')
+optimal_inequality = optimal_inequality.drop(
```

Comment:
would this work? could be clearer. also please add a comment about why this is necessary (not obvious to me)
```
optimal_inequality = optimal_inequality[optimal_inequality.funding_billions > 0]
```

Reply:
@MaxGhenis It makes the graphs look jumpy and weird, I'd include a screenshot if I could. 

@ngpsu22 It looks like we do this in the notebook as well, the code was just after writing the file, so I'll get rid of it in the notebook as it's now redundant